scu-de/src/fiskaltrust.Middleware.SCU.DE.SwissbitCloudV2/Services/SwissbitCloudV2ApiProvider.cs

```diff
@@ -0,0 +1,228 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using fiskaltrust.Middleware.SCU.DE.SwissbitCloudV2.Exceptions;
+using fiskaltrust.Middleware.SCU.DE.SwissbitCloudV2.Models;
+using Newtonsoft.Json;
+using System.Net.Http.Headers;
+using fiskaltrust.Middleware.SCU.DE.SwissbitCloudV2.Helpers;
+
+
+namespace fiskaltrust.Middleware.SCU.DE.SwissbitCloudV2.Services
+{
+    public sealed class SwissbitCloudV2ApiProvider : ISwissbitCloudV2ApiProvider, IDisposable
+    {
+        private const int EXPORT_TIMEOUT_MS = 18000 * 1000;
+
+        private readonly SwissbitCloudV2SCUConfiguration _configuration;
+        private readonly HttpClientWrapper _httpClient;
+        private readonly JsonSerializerSettings _serializerSettings;
+
+
+        public SwissbitCloudV2ApiProvider(SwissbitCloudV2SCUConfiguration configuration, HttpClientWrapper httpClient)
+        {
+            _configuration = configuration;
+            _httpClient = httpClient;
+            _serializerSettings = new JsonSerializerSettings
+            {
+                DefaultValueHandling = DefaultValueHandling.Ignore
+
+            };
+        }
+
+        public async Task<TransactionResponseDto> TransactionAsync(string transactionType, TransactionRequestDto transactionRequest)
+        {
+            var jsonPayload = JsonConvert.SerializeObject(transactionRequest, _serializerSettings);
+
+            var response = await _httpClient.PostAsync($"/api/v1/tse/{transactionType}", new StringContent(jsonPayload, Encoding.UTF8, "application/json"));
+            var responseContent = await response.Content.ReadAsStringAsync();
+            if (response.IsSuccessStatusCode)
```

Comment:
Generally applies for this whole file: 

I'm not sure if this is ever "false", right? Because the `HttpClientWrapper` will also check the success status of every response and also throws an exception. Potentially we could simplify this class here by getting rid of all the error handling, if I see correctly?

Reply:
Error handling is removed from the provider